sql/rowexec/rel_iters.go

```diff
@@ -1068,3 +1064,135 @@ func (ui *unionIter) Close(ctx *sql.Context) error {
 		return nil
 	}
 }
+
+type intersectIter struct {
+	lIter, rIter sql.RowIter
+	cached       bool
+	cache        map[uint64]int
```

Comment:
tests with these nested as join RHS will stress the caching, which will also maybe expose a memory leak in the Dispose() cleanup of these